src/main.rs

```diff
@@ -96,7 +97,12 @@ fn args_parser() -> Result<Args, pico_args::Error> {
             cache_max_age: args
                 .opt_value_from_fn("--cache-max-age", parse_max_age)?
                 .unwrap_or(default_cache_max_age),
+            free: args.free()?,
         };
+        if !args.free.is_empty() {
+            eprint_help();
+            panic!("Failed to parse arguments");
```

Comment:
Prefer returning an error instead of panicking (not sure why the other branch does that). In particular, we'd like to use `args.finish()` here but can't since we need `args` and can't move it into that method. The method does a very similar check and then returns the `Error::UnusedArgsLeft` variant.